### PR TITLE
boards: arm: Remove OPENSDA_FW cmake variables from nxp boards

### PR DESCRIPTION
**Pull request by MaureenHelm**
The OPENSDA_FW and LPCLINK_FW cmake variables are remnants of pre-west
days where we used an environment variable to set the desired debug host
tool, such as jlink or pyocd, based on which debug probe firmware was
loaded on the board. We now have two possible ways to do this, neither
of which requires the nxp-specific OPENSDA_FW or LPCLINK_FW variables:

1. Set standardized cmake runner variables when generating the build
   system:
   $ west build -- -DBOARD_FLASH_RUNNER=jlink -DBOARD_DEBUG_RUNNER=jlink

2. Use the west "--runner" argument with the debug and flash commands:
   $ west debug -r jlink

Remove the now unnecessary OPENSDA_FW ond LPCLINK_FW variables and
update board documentation accordingly.

A few boards (hexiwear_k64, mimxrt10{50,60,64}_evk) reordered
pyocd.board.cmake and jlink.board.cmake includes to preserve the default
runner when OPENSDA_FW was not explicitly set.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

### CODE REVIEW

**Review by mbolivar-nordic — APPROVED** [review]
Looks great, thanks for doing this.

**Review by henrikbrixandersen — APPROVED** [review]
LGTM.

**Review by dleach02 — APPROVED** [review]

File: boards/arm/frdm_k22f/board.cmake [hunk 1]
```diff
@@ -1,15 +1,5 @@
 # SPDX-License-Identifier: Apache-2.0
 
-set_ifndef(OPENSDA_FW daplink)
```
Comment (jfischer-phytec-iot) [inline boards/arm/frdm_k22f/board.cmake:3]: Does it not make jlink as default runner? That would break usual workflow. And probably CI tests on hardware?

Comment (mbolivar-nordic) [reply boards/arm/frdm_k22f/board.cmake:3]: No, pyocd should still be the default runner, because it's the first include below.

Comment (jfischer-phytec-iot) [reply boards/arm/frdm_k22f/board.cmake:3]: yes, sorry overlooked that.

**Review by galak — APPROVED** [review]

**Comment by galak** [comment]
Going to merge since CI will always timeout.

**Pull request merged**